### ISSUE TITLE
Badge request: OSSF Best Practices

### ISSUE THREAD

**Issue by miguelnietoa** [badges/shields#8399]
:clipboard: **Description**

[OpenSSF Best Practices Badge](https://bestpractices.coreinfrastructure.org/en) is used to show the level of security practices implemented in your open-source project.

> The [Open Source Security Foundation (OpenSSF)](https://openssf.org/) Best Practices badge is a way for Free/Libre and Open Source Software (FLOSS) projects to show that they follow best practices. Projects can voluntarily self-certify, at no cost, by using this web application to explain how they follow each best practice. The OpenSSF Best Practices Badge is inspired by the many badges available to projects on GitHub. Consumers of the badge can quickly assess which FLOSS projects are following best practices and as a result are more likely to produce higher-quality secure software.

- **Which service is this badge for:** 
  [OpenSSF Best Practices](https://bestpractices.coreinfrastructure.org/en)

- **What sort of information should this badge show?**
Examples in plain text:
    - `openssf best practices | in progress 19%`
    - `openssf best practices | passing`
    - `openssf best practices | silver`
    - `openssf best practices | gold`

:link: **Data**
Data is get from an API request like this:
`https://bestpractices.coreinfrastructure.org/projects/:projectId/badge.json`

https://bestpractices.coreinfrastructure.org/projects/6367/badge.json

```json
{
  "id": 6367,
  "name": "Elixir Stellar SDK",
  "updated_at": "2022-08-16T19:44:26.876Z",
  "badge_level": "passing",
  "tiered_percentage": 109
}
```

- **Is there a public API?** Yes.
- **Does the API require an API key?** No.
- **Link to the API documentation.** https://github.com/coreinfrastructure/best-practices-badge/blob/main/doc/api.md

:microphone: **Motivation**

This feature should be implemented since lately it is becoming popular to add these kinds of badges related to security practices and supply chain to the README of open source projects.

I have noticed that the badge service already exists for OSSF Scorecard, but not yet for OSSF Best Practices, so I think is nice to have this badge available too!

**Comment by miguelnietoa** [comment]
I would like to work on implementing this! 😄

**Comment by chris48s** [comment]
Are there any docs for this API. Is there any rate limit on it we need to be aware of?

**Comment by miguelnietoa** [comment]
Yes, @chris48s 

**API documentation:**
https://github.com/coreinfrastructure/best-practices-badge/blob/main/doc/api.md

**About rate limits:**
> All BadgeApp requests are rate-limited. At the time of this writing, if you keep requests at 1 request/second or less for anything other than badge images you'll be fine. Badge image rates can be much higher (those are specially optimized).

**Comment by chris48s** [comment]
Thanks. Broadly this is the kind of service we'd typically accept a PR for.

If you stretch that rate limit over an hour or something, a badge calling that endpoint would have to get fairly popular before it becomes a problem for us.

It seems like with a rate limit of 1 per sec on json requests (which `/badge.json` would be subject to) measured on quite a tight interval (based on https://github.com/coreinfrastructure/best-practices-badge/pull/1478 it looks like this is measured over a 15 second interval), this would not have to get _that_ popular before we start hitting rate limits and serving broken badges at busy times of day. We could do a certain amount by setting a longer than usual `max-age` to serve more traffic from downstream cache and avoid repeated requests for hits on the same badge URL but would only take one person somewhere putting a bunch of these things for different projects on the same page (inconsiderate but outside our control) to get us throttled.

**Comment by calebcartwright** [comment]
I kept coming to this and getting detoured, but I feel like this is a possible duplicate of our existing CII Best Practices badge?

https://img.shields.io/cii/percentage/29
![](https://img.shields.io/cii/percentage/29)

**Comment by miguelnietoa** [comment]
Hi @calebcartwright 🤝  
Yes, It is the same badge that I'm requesting to add. 

I have no idea why when searching for the badge by name directly on the shields.io site it does not appear. 🤔 
![image](https://user-images.githubusercontent.com/39246879/193147976-8f022667-f276-424f-bc9a-5f0c1579968e.png)

Thanks for the research about it! 😄

**Comment by chris48s** [comment]
I've submitted https://github.com/badges/shields/pull/8473 which fixes the search. I'm going to close this and #8400
